### ISSUE TITLE
Support for go executables

### ISSUE THREAD

**Issue by farimani** [GJDuck/e9patch#23]
Hi,

Are go executables supported?

 ```
package main      

import "fmt"

func main() {
       fmt.Println("hello world!")
}                                                                                               

```

`go build hello.go`

`./e9tool -O0 -M 'asm=jmp.*' -A print hello`

A couple of jump instructions are printed followed by segfault.

**Comment by GJDuck** [comment]
The first problem is that golang uses `PT_NOTE` to store data.  This can be fixed with some additional magic options:

    ./e9tool -O0 -M 'asm=jmp.*' -A print hello --option --loader-phdr=stack

The rewritten binary should get a lot further.  However, it will still crash when I try it:

    fatal error: unknown pc
    ...

This kind of error could occur if the golang runtime is doing something like parsing parts of the `.text` section during execution, which is fundamentally incompatible with static binary rewriting.

I had a look into this.  It seems that `runtime.rewindmorestack()` parses parts of the `.text` section, and this directly leads to the above runtime error when the corresponding bytes have been patched.  It might be possible to work around the problem by not modifying any instruction that golang parses, such as unconditional jumps.  This is a bit difficult to do currently, since E9Patch may modify neighbouring instructions for tactics T1/T2.

TL;DR

The golang runtime parses the `.text` section during execution, which is not compatible with static binary rewritng.

**Comment by farimani** [comment]
Thank you for this. Using the stack option and patching 'asm=syscall' instead of jump works. But I had to roll back to a release prior to your PE merge. I will open an issue with details for you.
